test.js

```diff
@@ -75,3 +74,58 @@ test('.stringify() should stringify srcset', t => {
 		'banner-HD.jpeg 2x, banner-phone.jpeg 100w'
 	);
 });
+
+const invalidStrings = [
+	'banner.jpeg, fallback.jpeg', // Multiple fallback images
+	'banner-phone-HD.jpg 100w 2x', // Multiple descriptors
+	'banner-HD.jpeg 2x, banner.jpeg 2x', // Multiple images with the same descriptor
```

Comment:
I think some of the changes in this PR can be considered breaking changes. I can understand a loose mode, but I'm not sure when a strict mode would be helpful? IMHO, we should try to follow the resolution logic of the browser as much as possible. For instance, Chrome don't throw any error when there are multiple images with the same descriptor, instead it simply select the first one.

A _stricter_ logic can be implemented in a separated function `validate` instead, so that we can simplify the source code here, and also allowing users to have more controls if needed.

At the very least, I don't think we should make this _strict_ mode enabled by default. Users with valid (valid as in compatible with most browsers) code in the browsers might experience unexpected errors when using this library.

That's just my 2 cents though. Curious to hear your thoughts :)

Reply:
> I don't think the strict mode should be enabled by default

Ok, that's easy enough to change. 

As for what browsers do, I did a bit of testing:
* Chrome and Firefox will both accept `100h 100w` - the current version of the library will throw an error on this
* Chrome logs errors on some invalid image candidates:
  * `Failed parsing 'srcset' attribute value since it has an 'h' descriptor and no 'w' descriptor.`
  * `Failed parsing 'srcset' attribute value since it has multiple 'x' descriptors or a mix of 'x' and 'w'/'h' descriptors.`
* Chrome silently ignores errors where multiple image candidates have the same descriptors (it just never uses any but the first image for any descriptor.)
* Firefox never logs any errors
* They have some difference in *which* image they choose - Chrome seems to prefer 2x images even on 1x displays, whereas Firefox is better at matching the image to the display density. (Chrome's behavior could be because I had a different display that was 2x attached. I didn't test it with only a 1x display.)